Sources/SourceKit/sourcekitd/SwiftCommand.swift

```diff
@@ -0,0 +1,110 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SKSupport
+import LanguageServerProtocol
+import Foundation
+
+/// The set of known Swift commands.
+///
+/// All commands from the Swift LSP should be listed here.
+public let builtinSwiftCommands: [String] = []
+
+/// A `Command` that should be executed by Swift's language server.
+public protocol SwiftCommand: Codable, Hashable {
+  var title: String { get set }
+  static var swiftCommandIdentifier: String { get }
+}
+
+extension SwiftCommand {
+  public static var identifier: String {
+    return Command.swiftCommandIdentifierPrefix + Self.swiftCommandIdentifier
+  }
+
+  /// Converts this `SwiftCommand` to a generic LSP `Command` object.
+  public func asCommand() throws -> Command {
```

Comment:
You can see how this gets used here:
https://github.com/apple/sourcekit-lsp/pull/70/files#diff-6175f3a32db6757aaeb5b49a98bab346R949